scripts/examples/gn_asr_example.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+set -e
+# Build script for GN ASR examples GitHub workflow.
+source "$(dirname "$0")/../../scripts/activate.sh"
+
+# Install required software
+
+set -x
+env
+
+#default
+if [ -z "$ASR_IC" ]; then
+    echo "ASR_IC not set! default asr582x"
+    export ASR_IC=asr582x
+fi
+if [ -z "$ASR_BOARD" ]; then
+    if [ "$ASR_IC" = "asr582x" ]; then
+        echo "ASR_BOARD not set! default ASR582X"
+        export ASR_BOARD=ASR582X
+    elif [ "$ASR_IC" = "asr595x" ]; then
+        echo "ASR_BOARD not set! default ASR595X"
+        export ASR_BOARD=ASR595X
+    fi
+fi
+if [ -z "$TOOLCHAIN_PATH" ]; then
+    echo "TOOLCHAIN_PATH not set! set default"
+    if [ "$ASR_IC" = "asr582x" ]; then
+        export TOOLCHAIN_PATH=/workspace/compiler/gcc-arm-none-eabi-9-2019-q4-major/bin/
```

Comment:
what ensures that `/workspace` is a valid path?
Why not just error out if toolchain is not set?